101-app-configuration-store-kv/azuredeploy.json

```diff
@@ -0,0 +1,55 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "configStoreName": {
+            "type": "string",
+            "metadata": {
+                "description": "Specifies the name of the App Configuration store."
+            }
+        },
+        "keyValueName": {
+            "type": "string",
+            "defaultValue": "myKey$myLabel",
+            "metadata": {
+                "description": "Specifies the name of the key-value resource. The name is a combination of key and label with $ as delimiter."
+            }
+        }
+    },
+    "resources": [
+        {
+            "name": "[parameters('configStoreName')]",
+            "type": "Microsoft.AppConfiguration/configurationStores",
+            "apiVersion": "2020-07-01-preview",
+            "location": "[resourceGroup().location]",
+            "sku": {
+                "name": "standard"
+            },
+            "resources": [
+                {
+                    "name": "[parameters('keyValueName')]",
+                    "type": "keyValues",
+                    "apiVersion": "2020-07-01-preview",
+                    "dependsOn": [
+                        "[parameters('configStoreName')]"
+                    ],
+                    "properties": {
+                        "value": "Hello World!",
+                        "contentType": "the-content-type",
+                        "tags": {
+                            "tag1": "tag-value-1",
+                            "tag2": "tag-value-2"
+                        }
+                    }
+                }
+                // Add multiple key-values to the new App Configuration store by defining additional keyValues resources here.
```

Comment:
a copy loop should be used if multiple values are required, we should probably add that to the sample rather than leave it as an exercise to the reader (it seems like it will be the common scenario

Reply:
Using copy loop is a great solution. But I think it's for pro users. I'm aware of copy loop but it still took me 30 mins to digest the syntax. 

I'm concerned that the learning curve might be too deep for beginners. Without fully going through this [doc](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/copy-resources), I'm afraid most users might have trouble understanding our template.

Reply:
I tried to write one as below. I think it's a bit difficult to read. @zhenlan What do you think?

```
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "configStoreName": {
            "type": "string",
            "defaultValue": "zhijzhao-prod",
            "metadata": {
                "description": "Specifies the name of the App Configuration store."
            }
        },
        "keyValueNames": {
            "type": "array",
            "defaultValue": [ "myKey$myLabel", "myKey2$myLabel2" ],
            "metadata": {
                "description": "Specifies the name of the key-value resource. The name is a combination of key and label with $ as delimiter."
            }
        },
        "keyValueValues": {
            "type": "array",
            "defaultValue": [ "Hello World 1", "Hellow World 2" ]
        }
    },
    "resources": [
        {
            "name": "[parameters('configStoreName')]",
            "type": "Microsoft.AppConfiguration/configurationStores",
            "apiVersion": "2020-07-01-preview",
            "location": "[resourceGroup().location]",
            "sku": {
                "name": "standard"
            }
        },
        {
            "name": "[concat(parameters('configStoreName'), '/', parameters('keyValueNames')[copyIndex()])]",
            "type": "Microsoft.AppConfiguration/configurationStores/keyValues",
            "apiVersion": "2020-07-01-preview",
            "dependsOn": [
                "[parameters('configStoreName')]"
            ],
            "properties": {
                "value": "[parameters('keyValueValues')[copyIndex()]]",
                "contentType": "the-content-type",
                "tags": {
                    "tag1": "tag-value-1",
                    "tag2": "tag-value-2"
                }
            },
            "copy": {
                "name": "keyValuesCopy",
                "count": "[length(parameters('keyValueNames'))]"
            }
        }
    ],
    "outputs": {
        "reference-value": {
            "value": "[reference(resourceId('Microsoft.AppConfiguration/configurationStores/keyValues', parameters('configStoreName'), parameters('keyValueNames')[0]), '2020-07-01-preview').value]",
            "type": "string"
        }
    }
}
```

Reply:
copy loop is a common practice in a template deployment so the more we can do to help users take advantage of the practice the better...  Your sample will help educate more users.

Reply:
@ZhijunZhao I like the copy loop example you have above. It's useful when people need to create multiple key-values that share the same metadata (content-type, tags, etc.). The example you put up is not hard to understand to me. In the documentation, we can add a sentence or two to explain what we are using and a link to the copy loop doc. It should help to build the context for people who are not familiar with the concept.

@bmoore-msft good suggestion. Thank you.